### PR TITLE
Add game replay saving, add checkpoint callback to save some replays along with the models

### PR DESCRIPTION
**Pull request by glmcdona**
By default saves 5 replay matches (itself against itself) every 100K steps during training along side the model. Saved to `.\models\`.

Adds:
* Game() ability to write replay outputs.
* SaveReplayAndModelCallback callback that can be used in training to save both model checkpoints along-side replays. Useful for examining the behavior that your model is learning.
* Change example training script default param n_steps to be the same as batch size.
* Action sequences now takes additional argument pointing to current game for better programmatic actions. I think this feature is unused, but will cause a failure if anyone else is using actions sequences unfortunately.

### CODE REVIEW

**Comment by glmcdona** [comment]
Note, I've done more testing and this isn't ready to merge yet. It seems to have some bugs now that I tested more replays, working on preparing a fix now.

**Comment by nosound2** [comment]
Great idea with replays. Can be nice to have `seed` in the replay filename. Will it be possible to reproduce the replay with a manual run? Similarly to how it is reproduced in tests.
